### PR TITLE
[Intel MKL] Enabling quantized pooling ops with native format

### PR DESCRIPTION
**Pull request by mahmoud-abuzaina**

### CODE REVIEW

File: tensorflow/core/kernels/mkl/mkl_quantized_pooling_ops_test.cc [hunk 1]
```diff
@@ -109,43 +116,54 @@ TEST_F(QuantizedPoolingTest, SmallAveragePooling) {
...
 
   TF_ASSERT_OK(RunOpKernel());
 
   const Tensor& output = *GetOutput(0);
-  const Tensor& mkl_shape_tensor = *GetOutput(3);
-  ConvMklToTF conv_comp;
   Tensor output_quantized;
-  conv_comp.ConvertMKL2TF<quint8>(DT_QUINT8, output, mkl_shape_tensor,
-                                  output_quantized);
+  if (!NativeFormatEnabled()) {
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl/mkl_quantized_pooling_ops_test.cc:129]: This part can use the refactored function [mentioned in #45108](https://github.com/tensorflow/tensorflow/pull/45108#discussion_r549582571).

File: tensorflow/core/kernels/mkl/mkl_quantized_pooling_ops_test.cc [hunk 2]
```diff
@@ -177,23 +195,28 @@ TEST_F(QuantizedPoolingTest, SmallMaxPooling) {
...
+                                    output_quantized);
+  }
 
   const float output_min = GetOutput(1)->flat<float>()(0);
   const float output_max = GetOutput(2)->flat<float>()(0);
-  Tensor output_float =
-      QuantizedTensorToFloat<quint8>(output_quantized, output_min, output_max);
+  Tensor output_float = QuantizedTensorToFloat<quint8>(
+      NativeFormatEnabled() ? output : output_quantized, output_min,
+      output_max);
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl/mkl_quantized_pooling_ops_test.cc:219]: This part can also call the [refactored function mentioned in #45108](https://github.com/tensorflow/tensorflow/pull/45108#discussion_r549582571)

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR and I'm so sorry for the delay! I have minor comments.

File: tensorflow/core/kernels/mkl/mkl_quantized_pooling_ops_test.cc (hunk 1 above)
Comment (mahmoud-abuzaina) [reply tensorflow/core/kernels/mkl/mkl_quantized_pooling_ops_test.cc]: I used the refactored function form #45108.

File: tensorflow/core/kernels/mkl/mkl_quantized_pooling_ops_test.cc (hunk 2 above)
Comment (mahmoud-abuzaina) [reply tensorflow/core/kernels/mkl/mkl_quantized_pooling_ops_test.cc]: I used the refactored function form #45108.

**Comment by mahmoud-abuzaina** [comment]
Thank you for the comments. I addressed them.

File: tensorflow/core/kernels/mkl/mkl_quantized_pooling_ops_test.cc [hunk 3]
```diff
@@ -125,20 +104,12 @@ TEST_F(QuantizedPoolingTest, SmallAveragePooling) {
...
-                                    output_quantized);
-  }
-
   const float output_min = GetOutput(1)->flat<float>()(0);
   const float output_max = GetOutput(2)->flat<float>()(0);
-  Tensor output_float = QuantizedTensorToFloat<quint8>(
-      NativeFormatEnabled() ? output : output_quantized, output_min,
-      output_max);
-
+  const Tensor* mkl_shape_tensor =
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl/mkl_quantized_pooling_ops_test.cc:109]: Nit: Can we add `_ptr` to the name? Same for line 180.
```suggestion
  const Tensor* mkl_shape_tensor_ptr =
```

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the changes! I have one minor nit.

File: tensorflow/core/kernels/mkl/mkl_quantized_pooling_ops_test.cc (hunk 3 above)
Comment (mahmoud-abuzaina) [reply tensorflow/core/kernels/mkl/mkl_quantized_pooling_ops_test.cc]: Sure. Done.

**Comment by mahmoud-abuzaina** [comment]
I have addressed the comment. Thank you!

**Review by penpornk — APPROVED** [review]
Thank you again for the PR!

**Comment by penpornk** [comment]
Will re-tag `ready-to-pull` after #45107 and #45108 are merged.

**Review by penpornk — APPROVED** [review]
Thank you for the PR and sorry for the delay!

**Review by gbaned — APPROVED** [review]

**Pull request merged**